include/exadg/aero_acoustic/calculators/source_term_calculator.h

```diff
@@ -87,6 +95,25 @@ class SourceTermCalculator
 
     Number rho = static_cast<Number>(data.density);
 
+    // In case we blend in the source term we check, if the scaling is space dependent. Only in that
+    // case we have to evaluate the function in every equadrature point. Otherwise the scaling
+    // purely temporal and constant during this function.
+    if(data.blend_in)
+      AssertThrow(data.blend_in_function != nullptr,
+                  dealii::ExcMessage("No blend-in function provided."));
+
+    bool const space_dependent_scaling =
+      data.blend_in_function != nullptr ? data.blend_in_function->varies_in_space() : false;
+    Number const pure_temporal_scaling_factor =
+      data.blend_in_function != nullptr ? data.blend_in_function->compute_time_factor() : 1.0;
```

Comment:
For the function `compute_time_factor()` to be evaluated correctly, where do you call `function->set_time()` in the code? As written in comment https://github.com/exadg/exadg/pull/644/files#r1528630370, my current understanding is that we should add `double const time` to the interface of this function. However, I might misunderstand something.

Reply:
[ef52a6e](https://github.com/exadg/exadg/pull/644/commits/ef52a6ec89f956fff89d3ecea0b015c02f4e9a40)